hibernate-core/src/main/java/org/hibernate/boot/model/naming/NamingHelper.java

```diff
@@ -135,17 +138,36 @@ public String generateHashedConstraintName(
 	 */
 	public String hashedName(String name) {
 		try {
-			final MessageDigest md5 = MessageDigest.getInstance( "MD5" );
-			md5.reset();
-			md5.update( charset != null ? name.getBytes( charset ) : name.getBytes() );
-			final BigInteger bigInt = new BigInteger( 1, md5.digest() );
-			// By converting to base 35 (full alphanumeric), we guarantee
-			// that the length of the name will always be smaller than the 30
-			// character identifier restriction enforced by a few dialects.
-			return bigInt.toString( 35 );
+			return hashWithAlgorithm(name, "MD5");
 		}
-		catch ( NoSuchAlgorithmException | UnsupportedEncodingException e ) {
-			throw new HibernateException( "Unable to generate a hashed name", e );
+		catch (NoSuchAlgorithmException | UnsupportedEncodingException e) {
+			log.warnf("MD5 algorithm failed for hashedName, falling back to SHA-256: %s", e.getMessage());
```

Comment:
A `WARN` is probably too aggressive here.

Reply:
@gavinking I kept the log level as WARN so that people notice when the MD5 hashing constraint fails for some reason. If we set the level to INFO, it might go unnoticed. Would you like me to change it to INFO?

Reply:
Changed the log level to info.